tests/unit/lms/views/dashboard/api/assignment_test.py

```diff
@@ -140,6 +140,77 @@ def test_course_assignments(
             ]
         }
 
+    def test_course_assignments_filtered_by_assignment_ids(
+        self,
+        views,
+        pyramid_request,
+        assignment_service,
+        h_api,
+        db_session,
+        dashboard_service,
+        user_service,
+    ):
```

Comment:
There is a lot of code in this test that isn't directly connected to the change to filter by assignment IDs. It looks like you've got some future refactoring in-flight to reduce this in https://github.com/hypothesis/lms/pull/6437. What's the plan here?

Reply:
Merged #6437 into this one so we can look at them combined.

Reply:
For this particular test I changed the assertions to be the most relevant to the test case.

We could move more lines here to fixtures but there's a balance between verbosity in the test vs tightly coupled fixtures.